R/tbl-format.R

```diff
@@ -60,17 +60,39 @@ format_tbl <- function(
   force(x)
   num_colors(forget = TRUE)
 
-  setup <- tbl_format_setup(x,
-    width = width, ...,
+  setup_used <- FALSE
+
+  setup <- tbl_format_setup(
+    x,
+    width = width,
+    ...,
+    setup = {
+      setup_used <- TRUE
+      NULL
+    },
     n = n,
     max_extra_cols = max_extra_cols,
     max_footer_lines = max_footer_lines,
     focus = attr(x, "pillar_focus")
   )
 
-  header <- tbl_format_header(x, setup)
-  body <- tbl_format_body(x, setup)
-  footer <- tbl_format_footer(x, setup)
+  header <- transform(tbl_format_header(x, setup))
```

Comment:
What does `transform()` do?

Reply:
`transform` is `writeLines` when printing and `identity` when formatting. This is how we achieve incremental printing.

Adding a comment to that effect to the code.